sass/worker.js

```diff
@@ -0,0 +1,169 @@
+(function (factory) {
```

Comment:
It looks like this file was copied from the output of a TypeScript source file. Is there any reason this code can't also be TypeScript? It's kind of odd to copy code output like this. The code has `tslint` comments and also still references the TypeScript compiler.

Reply:
Yes, it was copied from the TS output.

The main reason to use JS instead of TS is to avoid depending on TS compiler. I agree that the comments are confusing.

I guess there are few options here:

- Copy worker.ts and depending on the TSC in this repro. Deleting the comments that references to TSC
- Write a roughly identical version of worker.js, make it more generic

Not sure which one is better.

Reply:
@nex3 any opinions on this? I would want to either use a TypeScript source file or use a cleaned-up JavaScript implementation rather than copying a build artifact.

Reply:
At a higher level, why are we implementing worker management in this repo to begin with? I'd say this should live in its own package—it's not Sass-specific at all.

To address the more direct question, I agree with Jeremy that checking in a generated file is probably not the best way to go. Do Angular's Bazel rules depend on the TS compiler? If so, I think it makes sense to do so for this as well.

Reply:
I agree that is not Sass-specific at all, but it is also hard to find/release a separate package for it (at least IMO).

I actually inlined all of the worker.js into sass_wrapper.js; it is ~100 lines of code that handles the buffer and arguments. IMHO It is fairly simple code that parses args and be compliant with bazel worker protocol. While it is unfortunate to duplicate these code, making these a generic package might be an overkill. Actually you will find many internal tools in google3 doing this (adding ~100 lines of code in different languages so that the tool can benefit from persistent worker).

Please let me know what you think.

Reply:
I'd really like to see this persistent worker code packaged and re-usable. You're right that it's not a lot of lines of code, but that code embeds a lot of expertise in the worker protocol, and if that protocol ever changes it would be a huge pain to update every usage of it. I'm sure https://github.com/bazelbuild/rules_typescript would love to be able to share that logic as well, for example.

Reply:
Please take another look -- Alex released a @bazel/worker package and I changed this commit to use that package.